api/src/services/graphql/utils/sanitize-gql-schema.ts

```diff
@@ -0,0 +1,102 @@
+import type { Relation, SchemaOverview } from '@directus/types';
+import { useLogger } from '../../../logger.js';
+
+/**
+ * Regex was taken from the spec
+ * https://spec.graphql.org/June2018/#sec-Names
+ */
+const GRAPHQL_NAME_REGEX = /^[_A-Za-z][_0-9A-Za-z]*$/;
+
+// Not an exhaustive list (doesnt include generated names)
+const GRAPHQL_RESERVED_NAMES = [
+	'Subscription',
+	'Query',
+	'Mutation',
+	'Int',
+	'Float',
+	'String',
+	'Boolean',
+	'DateTime',
+	'ID',
+	'uid',
+	'Point',
+	'PointList',
+	'Polygon',
+	'MultiPolygon',
```

Comment:
As far as I can tell we're missing the types added by Directus, like `JSON`, `Hash`, `Void`, we define in `graphql/types`. We can add the ones we currently support or we can change the types to a module level export and then autogenerate it based on their `name` property.

```suggestion
	'MultiPolygon',
	// Directus defined types
	'GraphQLBigInt',
	'Hash',
	'JSON',
	'Date',
	'GraphQLStringOrFloat',
	'Void',
```

Reply:
I'll add those few defined too but this is a static non-exhaustive list, generating other types or making this list "complete" is out of scope for this PR.